### ISSUE TITLE
Support negated pattern matching `!(...patterns)`

### ISSUE THREAD

**Issue by citycide** [citycide/glob#1]
The rest of the extended pattern matching forms are [already supported](https://citycide.github.io/glob/#syntax-extended-pattern-matching) but this one is actually not technically possible due to the current regex backend being used. `glob` needs to generate a regex negative lookahead of the form `(?!...)` which isn't implemented yet in [`regex`](https://github.com/nitely/nim-regex).

There's an existing issue over there for it: https://github.com/nitely/nim-regex/issues/6

I'm assuming both Nim's stdlib regex engines support it but they both wrap `pcre` which removes the pure Nim aspect of `glob`.

**Comment by timotheecour** [comment]
@citycide can user defined `filterDescend` be used for that? see https://github.com/citycide/glob/issues/8
eg:
```
import glob
import re # or nre
walkGlobKinds(".", filterDescend = dir=>not dir.matches(re"pattern1")).filter(a=>not a.path.matches(re"pattern2"))
```
or:
```
import glob
import regex
walkGlobKinds(".", filterDescend = dir=>dir.matches(re"!(...patterns1)")).filter(a=>a.path.matches(re"!(...patterns2)")
```

**Comment by citycide** [comment]
Update: while the linked issue has been closed, only single-character lookarounds are supported in `(?!...)` groups. Progress, but not enough IMO to enable the feature in `glob` yet.

**Comment by citycide** [comment]
Closed by https://github.com/citycide/glob/commit/83bdafc85ffdcc64004788ded4b94fe8ba06fb7c and released in [v0.11.0](https://github.com/citycide/glob/releases/tag/v0.11.0).
